.github/workflows/gh-release.yaml

```diff
@@ -16,7 +16,36 @@ jobs:
     needs: tests
 
     steps:
+      - name: Validate version in tag
+        # check that version matches Semantic Versioning
+        run: |
+          if ! [[ ${{ github.ref_name }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$|^[0-9]+\.[0-9]+\.[0-9]+\.dev[0-9]+$ ]]
+          then
+            echo "Tag has incorrect format. Valid formats are X.X.X or X.X.X.devX where X is any digit"
+            exit 1
+          fi
+      - name: Check tag and PyPi version
+        # Compare version of the tag with the latest PyPi published version
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
`Install dependencies` below does this same thing.  Let's move `Install dependencies` to the top and remove this line

Reply:
Actually it would be moving install dependencies and the checkout above this, as the dependencies uses the checkout